### ISSUE TITLE
OAuth 1.0 Trello issue

### ISSUE THREAD

**Issue by kaushik12048** [scribejava/scribejava#1064]
Since Trello API's only supports Oauth 1.0, recently it stopped working for Trello access token. (getRequestToken())
Failing to get access token.

Found similar issues related to it: https://community.developer.atlassian.com/t/error-when-requesting-trello-access-token-response-code-400-message-bad-request-body-null/44737

Also, this happens only for Trello. Twitter v1 API's using Oauth 1.0, seems to be work perfectly fine with ScribeJava.
Please fix this.

**Comment by kaushik12048** [comment]
@v0o0v @M-F-K Any update on this?

**Comment by kaushik12048** [comment]
@v0o0v  @M-F-K @softprops @luisdelarosa @fernandezpablo85 Can you please share any updates?

**Comment by s-gromov** [comment]
Hi @kaushik12048 
Unfortunately, Scribejava does not support OAuth2 for Trello at the moment.
But you can always create an API class yourself. You can take any API as an example and this wiki page as a guide
https://github.com/scribejava/scribejava/wiki/Custom-Apis#how-to-work-with-other-apis

Or hire any developer to do it for you.

I don't see any open Pull Request with support for OAuth2 in Trello. If you provide a Pull Request, it will be a good contribution for other Scribejava users

**Comment by kullfar** [comment]
Hi @kaushik12048
Unfortunately, Scribejava does not support OAuth2 for Trello at the moment.
But you can always create an API class yourself. You can take any API as an example and this wiki page as a guide
https://github.com/scribejava/scribejava/wiki/Custom-Apis#how-to-work-with-other-apis

Or hire any developer to do it for you.

I don't see any open Pull Request with support for OAuth2 in Trello. If you provide a Pull Request, it will be a good contribution for other Scribejava users

**Comment by kullfar** [comment]
Hi @kaushik12048
Unfortunately, Scribejava does not support OAuth2 for Trello at the moment.
But you can always create an API class yourself. You can take any API as an example and this wiki page as a guide
https://github.com/scribejava/scribejava/wiki/Custom-Apis#how-to-work-with-other-apis

Or hire any developer to do it for you.

I don't see any open Pull Request with support for OAuth2 in Trello. If you provide a Pull Request, it will be a good contribution for other Scribejava users

**Comment by kullfar** [comment]
https://github.com/scribejava/scribejava/wiki/FAQ#when-will-you-support-insert_api_here-

**Comment by kaushik12048** [comment]
@kullfar  Hi Kullfar, this is related to Oauth 1.0 api's only. Can we please check the issue for v1 api for Trello?
Fetching Access token is failing for Trello.

**Comment by kullfar** [comment]
Oh, I see, I got confused. Sorry. I'll reopen the issue so it stays in the queue. But I don't have an ETA for this ticket yet.

**Comment by M-F-K** [comment]
just for curiosity, I took a look at this. Seems like Trello now requires that the "allowed origin" to be specified. Read more about it here:
 
https://developer.atlassian.com/cloud/trello/guides/rest-api/authorization/#using-basic-oauth

The call OAuthAuthorizeToken also fails for me with the Trello example. 

I tried setting the Allowed origin to http://localhost:3000, and calling again, - no success :-(

Then I tried to copy the url generated by scribejava into Postman and add the return_url param like this:
https://trello.com/1/OAuthAuthorizeToken?oauth_token={MY_TOKEN}&return_url=http://localhost:3000
but no success, - I still get the **Invalid return_url. The return URL should match the application's allowed origins. Contact the developer.**

 it seems like scribejava works, but the problem seems to be specifying the correct URL's in the correct places ( the url part and and the "allowed" origin part at the Trello site ).  IF the return_url param really is needed in the OAuthAuthorizeToken call, then a small fix is needed (to at least the Trello example).

When I think about it, the last time i used Trello with scribejava (about 4 years ago), I didn't fill out the "allowed origin" at their site (and it worked nicely). The reason I know this is that I checked again today before I began hacking, and the list was empty. So hardening the security setup at Trello is probably the cause.

Also found this : https://community.developer.atlassian.com/t/invalid-return-url-on-rest-api-authentication-despite-set-allowed-origins/85294

Oh well, hope someone can use this info .

**Comment by kaushik12048** [comment]
@M-F-K  @kullfar  Can we please get this fixed?

**Comment by M-F-K** [comment]
> [@M-F-K](https://github.com/M-F-K) [@kullfar](https://github.com/kullfar) Can we please get this fixed?

Of course, - I think @kullfar will be happy to recieve a pullrequest,that fixes the problem, from you. Since this an opensource project everybody who has the time can pitch in =)   I read somewhere that he also do paid consultant work if time is an issue.
